### PR TITLE
make readme say more about what tsickle does

### PR DESCRIPTION
**Pull request by evmar**
We're no longer just a type-annotator, we do a lot of transformation!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,27 @@
-# Tsickle - TypeScript to Closure Annotator [![Build Status](https://travis-ci.org/angular/tsickle.svg?branch=master)](https://travis-ci.org/angular/tsickle)
+# Tsickle - TypeScript to Closure Translator [![Build Status](https://travis-ci.org/angular/tsickle.svg?branch=master)](https://travis-ci.org/angular/tsickle)
 
-Tsickle processes TypeScript and adds [Closure Compiler]-compatible JSDoc
-annotations. This allows using TypeScript to transpile your sources, and then
+Tsickle converts TypeScript code into a form acceptible to the [Closure Compiler].
```
Comment (kylecordes) [inline README.md]: acceptible -> acceptable

File: README.md [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
 information in Closure Compiler.
 
 [Closure Compiler]: https://github.com/google/closure-compiler/
 
+## What conversion means
+
+A (non-exhaustive) list of the sorts of transformations Tsickle applies:
+
+- inserts Closure-compatible JSDoc annotations on functions/classes/etc
+- converts ES6 modules into `goog.module`/`goog.provide`
```
Comment (mprobst) [inline README.md]: nit: / `goog.forwardDeclare`

Comment (kylecordes) [inline README.md]: How about mentioning here, that it performs this translation by means of common JS modules, hence it requires module:commonjs in the TSConfig?  I am working on a change to stop within explanatory error if that configuration is missing or incorrect.

File: README.md (hunk 1 above)
Comment (evmar) [inline README.md]: Thanks!

File: README.md (hunk 2 above)
Comment (evmar) [inline README.md]: I'm trying to stay high-level here so I will drop the goog.provide part

Comment (evmar) [inline README.md]: I'm trying to stay high-level here (and I think we can just fix this bug anyway)

Comment (kylecordes) [inline README.md]: @evmar For the sake of people coming in from outside, I think that sentence mentioning goog.* is fantastic. It is a critical thing to understand about what tsickle does and does not do.  If I were to give somebody a super concise high-level description of tsickle, I would probably say:

Tsickle extends the TypeScript compiler to emit JavaScript code containing Closure-Compiler-compatible JSDoc-like comment annotations translated from the original types. It  can emit either ES5 or ES2016 JavaScript output, but only in Closure "goog.module" modules, not in ES2016 modules.

Comment (evmar) [inline README.md]: Part of the point of tsickle is for users to need to know about goog.module modules. :)  (In principle if Closure Compiler improved its ES6 support we may not even do this transformation.)

The point of this wall of text is instead just to say "tsickle does some invasive code transformations, it doesn't just insert jsdoc" (which is what outsiders often think when they're not familiar with the guts).

Comment (kylecordes) [inline README.md]: So far in my experience (keeping in mind that I'm probably doing many things wrong, but in my defense there is not an abundance of documentation...) I have needed to be aware of how these module systems interoperate in Closure. By the way the documentation on that last bit is not particularly encouraging:

https://github.com/google/closure-compiler/wiki/JS-Modules#module-interoperation

However I am completely thrilled that making it moot is a goal - In this readme is a great improvement regardless of my nitpick :-)

**Pull request merged**